cmd/soroban-cli/src/commands/contract/build.rs

```diff
@@ -62,6 +62,20 @@ pub struct Cmd {
     /// Print commands to build without executing them
     #[arg(long, conflicts_with = "out_dir", help_heading = "Other")]
     pub print_commands_only: bool,
+    /// Add key-value to contract meta (adds the meta to the `contractmetav0` custom section)
+    #[arg(long, num_args=1, value_parser=parse_meta_arg, action=clap::ArgAction::Append, help_heading = "Metadata")]
+    pub meta: Vec<(String, String)>,
+}
+
+fn parse_meta_arg(s: &str) -> Result<(String, String), Error> {
+    let parts = s.splitn(2, '=');
```

Comment:
Should probably also check that there's only one `=` in the string

Reply:
`=` is a valid character in a header value, so it's allowed to occur more than once since only the first one we use as a delimiter on the key-value.

Reply:
Oops, thought this was about the rpc headers PR 😓. There's little reason to block people from using `=` in a meta value too.